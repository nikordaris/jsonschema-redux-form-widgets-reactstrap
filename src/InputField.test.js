import React from 'react';
import renderer from 'react-test-renderer';
import _inputFields from './index';
import { forEach, get } from 'lodash';
import { Provider } from 'react-redux';
import { reduxForm, reducer as formReducer } from 'redux-form';
import { createStore, combineReducers } from 'redux';

const { SingleSelectInput, Card, ...inputFields } = _inputFields;

const testInputFieldSnapshot = options => () => {
  const rootReducers = combineReducers({ form: formReducer });
  const store = createStore(rootReducers);
  const InputComponent = get(_inputFields, options.name);
  const WrappedComponent = reduxForm({ form: 'MyForm' })(InputComponent);
  const tree = renderer
    .create(
      <Provider store={store}>
        <WrappedComponent {...options} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
};

describe('Render InputFields', () => {
  forEach(inputFields, (InputComponent, name, idx) => {
    const schema = {
      id: name,
      title: name,
      type: 'string',
      meta: {
        vis: {
          ordinal: idx,
          component: name
        }
      }
    };
    it(
      `Should render ${name}`,
      testInputFieldSnapshot({ schema, name }, InputComponent)
    );
  });
});
