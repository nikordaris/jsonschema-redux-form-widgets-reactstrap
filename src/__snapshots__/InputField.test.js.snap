// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render InputFields Should render ColorInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="ColorInputField"
    id="ColorInputField-label"
  >
    ColorInputField
  </label>
  <input
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="input-0-0 form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="ColorInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="ColorInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    type="color"
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;

exports[`Render InputFields Should render DateInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="DateInputField"
    id="DateInputField-label"
  >
    DateInputField
  </label>
  <input
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="DateInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="DateInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    type="date"
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;

exports[`Render InputFields Should render DateTimeInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="DateTimeInputField"
    id="DateTimeInputField-label"
  >
    DateTimeInputField
  </label>
  <input
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="DateTimeInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="DateTimeInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    type="datetime-local"
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;

exports[`Render InputFields Should render EmailInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="EmailInputField"
    id="EmailInputField-label"
  >
    EmailInputField
  </label>
  <input
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="EmailInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="EmailInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    type="email"
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;

exports[`Render InputFields Should render NumberInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="NumberInputField"
    id="NumberInputField-label"
  >
    NumberInputField
  </label>
  <input
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="NumberInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="NumberInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    type="number"
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;

exports[`Render InputFields Should render PasswordInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="PasswordInputField"
    id="PasswordInputField-label"
  >
    PasswordInputField
  </label>
  <input
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="PasswordInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="PasswordInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    type="password"
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;

exports[`Render InputFields Should render TextAreaInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="TextAreaInputField"
    id="TextAreaInputField-label"
  >
    TextAreaInputField
  </label>
  <textarea
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="TextAreaInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="TextAreaInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;

exports[`Render InputFields Should render TextInputField 1`] = `
<div
  anyTouched={false}
  array={
    Object {
      "insert": [Function],
      "move": [Function],
      "pop": [Function],
      "push": [Function],
      "remove": [Function],
      "removeAll": [Function],
      "shift": [Function],
      "splice": [Function],
      "swap": [Function],
      "unshift": [Function],
    }
  }
  asyncValidate={[Function]}
  autofill={[Function]}
  blur={[Function]}
  change={[Function]}
  className="form-group"
  clearAsyncError={[Function]}
  clearSubmit={[Function]}
  clearSubmitErrors={[Function]}
  destroy={[Function]}
  error={undefined}
  handleSubmit={[Function]}
  initialValues={undefined}
  initialize={[Function]}
  initialized={false}
  invalid={false}
  pure={true}
  reset={[Function]}
  submit={[Function]}
  submitSucceeded={false}
  touch={[Function]}
  triggerSubmit={undefined}
  untouch={[Function]}
  valid={true}
  warning={undefined}
>
  <label
    className=""
    htmlFor="TextInputField"
    id="TextInputField-label"
  >
    TextInputField
  </label>
  <input
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    className="form-control"
    clearAsyncError={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    error={undefined}
    handleSubmit={[Function]}
    id="TextInputField"
    initialValues={undefined}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    name="TextInputField"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    pure={true}
    required={false}
    reset={[Function]}
    submit={[Function]}
    submitSucceeded={false}
    touch={[Function]}
    triggerSubmit={undefined}
    type="text"
    untouch={[Function]}
    valid={true}
    value=""
    warning={undefined}
  />
</div>
`;
